packages/webamp-modern-2/src/skin/Bitmap.ts

```diff
@@ -136,21 +136,11 @@ export default class Bitmap {
     if (this._canvas == null) {
       assert(this._img != null, "Expected bitmap image to be loaded");
       this._canvas = document.createElement("canvas");
-      this._canvas.width = this.getWidth();
-      this._canvas.height = this.getHeight();
+      this._canvas.width = this.getWidth() || this._img.width;
+      this._canvas.height = this.getHeight() || this._img.height;
       const ctx = this._canvas.getContext("2d");
       // https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage
-      ctx.drawImage(
-        this._img,
-        this._x,
-        this._y,
-        this.getWidth(),
-        this.getHeight(),
-        0,
-        0,
-        this.getWidth(),
-        this.getHeight()
-      );
+      ctx.drawImage(this._img, -this._x, -this._y);
```

Comment:
Cool!

Reply:
Yeah! it was several complicated fail there, before I got this into true implementation.